logic/src/commonTest/kotlin/com/wire/kalium/logic/data/message/MessageRepositoryTest.kt

```diff
@@ -153,6 +157,74 @@ class MessageRepositoryTest {
         }
     }
 
+    @Test
+    fun givenAMessage_whenSendingReturnsSuccess_thenCorrectConversationMessageOptionShouldBeSent() = runTest {
+        val messageEnvelope = MessageEnvelope(TEST_CLIENT_ID, listOf())
+        val mappedId: NetworkQualifiedId = TEST_NETWORK_QUALIFIED_ID_ENTITY
+        val timestamp = TEST_DATETIME
+
+        val (arrangement, messageRepository) = Arrangement()
+            .withMappedApiModelId(mappedId)
+            .withSuccessfulMessageDelivery(timestamp)
+            .arrange()
+
+        messageRepository.sendEnvelope(TEST_CONVERSATION_ID, messageEnvelope, MessageTarget.Conversation)
+            .shouldSucceed()
+
+        val expectedParameters = MessageApi.Parameters.QualifiedDefaultParameters(
+            sender = messageEnvelope.senderClientId.value,
+            recipients = mapOf(),
+            true,
+            MessagePriority.HIGH,
+            false,
+            messageEnvelope.dataBlob?.data,
+            MessageApi.QualifiedMessageOption.ReportAll
+        )
+
+        verify(arrangement.messageApi)
+            .suspendFunction(arrangement.messageApi::qualifiedSendMessage)
+            .with(eq(expectedParameters), anything())
+    }
+
+    @Test
+    fun givenAMessage_whenSendingReturnsSuccess_thenCorrectClientsMessageOptionShouldBeSent() = runTest {
+        val messageEnvelope = MessageEnvelope(TEST_CLIENT_ID, listOf())
+        val mappedId: NetworkQualifiedId = TEST_NETWORK_QUALIFIED_ID_ENTITY
+        val timestamp = TEST_DATETIME
+
+        val (arrangement, messageRepository) = Arrangement()
+            .withMappedApiModelId(mappedId)
+            .withSuccessfulMessageDelivery(timestamp)
+            .arrange()
+
+        messageRepository.sendEnvelope(
+            TEST_CONVERSATION_ID,
+            messageEnvelope,
+            MessageTarget.Client(
+                recipients = listOf(
+                    Recipient(
+                        id = TEST_USER_ID,
+                        clients = listOf(TEST_CLIENT_ID)
+                    )
+                )
+            )
+        ).shouldSucceed()
+
+        val expectedParameters = MessageApi.Parameters.QualifiedDefaultParameters(
+            sender = messageEnvelope.senderClientId.value,
+            recipients = mapOf(),
+            true,
+            MessagePriority.HIGH,
+            false,
+            messageEnvelope.dataBlob?.data,
+            MessageApi.QualifiedMessageOption.IgnoreAll
+        )
```

Comment:
Should we really assert things like Message Priority, data blob, etc.? These are independent from the message targets.
When we change the message priority logic, this test will fail without any reason.

I suggest to, instead, use a `matching {}` lambda to assert only the parameters we actually need:


```
        verify(arrangement.messageApi)
            .suspendFunction(arrangement.messageApi::qualifiedSendMessage)
            .with(matching {
            it.recipients.isEmpty() && it.options = MessageApi.QualifiedMessageOption.IgnoreAll
         }, anything())
```

Same thing for the test above.

Reply:
Aha! Nice, I was not aware that you could work with it like that. Thanks for the heads up 🙏🏻